tasties_app/models.py

```diff
@@ -1,3 +1,80 @@
-# from django.db import models
+from django.db import models
+from django.utils import timezone
+from django.contrib.auth.models import User
+from django.core.validators import MinValueValidator, MinLengthValidator, MaxValueValidator
+from datetime import datetime
 
-# Create your models here.
+
+class Category(models.Model):
+    category_name = models.CharField(max_length=16, unique=True)
+
+
+class Recipe(models.Model):
+    title = models.CharField(max_length=64, validators=[MinLengthValidator(1)], unique=True)
+    author_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    categories = models.ManyToManyField(Category)
+    description = models.CharField(max_length=512, validators=[MinLengthValidator(1)])
+    directions = models.CharField(max_length=65536, validators=[MinLengthValidator(1)])
+    publication_date = models.DateField(default=timezone.now)
+    minutes_to_make = models.IntegerField(validators=[MinValueValidator(1)])
+    recipe_picture = models.ImageField(upload_to='recipe_pictures')
+
+    def __str__(self):
+        return self.title
+
+
+class Ingredient(models.Model):
+    recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
+    amount = models.DecimalField(max_digits=10, decimal_places=2, validators=[MinValueValidator(0.0001)])
+    measurement_unit = models.TextChoices('measurement_unit', 'ML FLOZ TSP OZ CUP GRAM WHOLE')
+    description = models.CharField(max_length=64, validators=[MinLengthValidator(1)])
+
+
+class Comment(models.Model):
+    author_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
+    publication_date = models.DateField(default=timezone.now, blank=True)
+    comment_text = models.CharField(max_length=512, validators=[MinLengthValidator(1)])
+
+    def __str__(self):
+        return self.comment_text
+
+    @staticmethod
+    def show_all_comment_author(a_id):
+        user = User.objects.get(username=a_id)
+        Obje = Comment.objects.filter(author_id=user)
+        lis = [i.comment_text for i in Obje]
+        return lis
+
+    @staticmethod
+    def add_comment(a_id, r_id, text):
+        dat = datetime.today().strftime('%Y-%m-%d')
```

Comment:
It is not necessary to explicitly define a date variable ("dat"). Notice that in the definition of the field publication_date in the Comment class, we give it a default value of "timezone.now" - this gives us the exact functionality we want, which is to the the current date as the publication_date.

Reply:
you right but also wee need to fill this field by string in type 'yyyy-mm-dd',so I used datetime i know is no matter the real time but for more good features .